ui/src/tree/AzExtTreeItem.ts

```diff
@@ -44,6 +46,14 @@ export abstract class AzExtTreeItem implements types.AzExtTreeItem {
         return this._temporaryDescription || this.description;
     }
 
+    public get id(): string {
+        return this._id || this.label;
```

Comment:
Let's not do `|| this.label` here since it's handled elsewhere and since this accessor may get overridden by child classes.

The return type should be `string | undefined` like all the other optional properties